e2e/helpers/wallet.helper.ts

```diff
@@ -23,9 +25,73 @@ export type UTXO = {
 
 export class WalletHelper {
     private root: BIP32Interface;
-
+    private bitcoinRPCUtil: BitcoinRPCUtil;
+    private spendableAmount = 0;
```

Comment:
For the second case, halving occurs every 210,000 blocks, which is why the line `private lastAcceptableBlock = 209_999;` is set to ensure that we use only up to 209,999 blocks. This should be sufficient for any variation of end-to-end tests. Additionally, the `initializeWallet()` function includes the logic `this.currentBlockCount += 101;` to ensure the first 50 BTC is spendable, working alongside the logic `this.spendableAmount += 50;` in the `mineBlock` function." `initializeWallet()`  contains the logic `this.currentBlockCount += 101;` which ensures the first 50 btc is spendable, in conjuction with the logic  `this.spendableAmount += 50;` in the `mineBlock` function.

Reply:
For the first reason, since the 50 BTC is determined based on the knowledge of `lastAcceptableBlock`, it’s unclear how this would contribute to an increased maintenance burden.

Reply:
i believe this approach is more error-prone and complex compared to fetching the balance directly from the node. that's one reason which contribute to increase maintenance burden. because new devs wouldn't easily understand what's going on. so fetching the balance directly from node is the right approach

Reply:
> For the first reason, since the 50 BTC is determined based on the knowledge of `lastAcceptableBlock`, it’s unclear how this would contribute to an increased maintenance burden.

Perfect example of the maintenance burden is the code you wrote. It's bugged because your base assumption that halving occurs every 210,000 blocks. On regtest, halving happens every 150 blocks.

Reply:
hance the increase bug surface area and maintenance burden